Samples/vshadow/src/vssclient.cpp

```diff
@@ -124,6 +124,53 @@ HRESULT VssClient::IsVolumeSupported(VSS_PWSZ pwszVolumeName, BOOL* pbSupportedB
     return m_pVssObject->IsVolumeSupported(GUID_NULL, pwszVolumeName, pbSupportedByThisProvider);
 }
 
+HRESULT VssClient::SetShadowSpaceUnlimited(VSS_PWSZ volume_name)
+{
+    FunctionTracer ft(DBG_INFO);
+
+    const HRESULT ALREADY_SET = 0x20000000;
+    HRESULT result = ALREADY_SET;
+
+    CComPtr<IVssDifferentialSoftwareSnapshotMgmt> vss_diff_mgmt;
+    CComPtr<IVssSnapshotMgmt> vss_mgmt;
+
+    CHECK_COM(vss_mgmt.CoCreateInstance(
+        CLSID_VssSnapshotMgmt,
+        nullptr,
+        CLSCTX_ALL));
+
+    CHECK_COM(vss_mgmt->GetProviderMgmtInterface(
+        VSS_SWPRV_ProviderId,
+        IID_IVssDifferentialSoftwareSnapshotMgmt,
+        reinterpret_cast<IUnknown**>(&vss_diff_mgmt)));
+
+    CComPtr<IVssEnumMgmtObject> enum_vss;
+    CHECK_COM(vss_diff_mgmt->QueryDiffAreasForVolume(volume_name, &enum_vss));
+
+    VSS_MGMT_OBJECT_PROP class_object;
+    VSS_DIFF_AREA_PROP& diff_area = class_object.Obj.DiffArea;
```

Comment:
Avoid reusing these variables in each loop iteration. Move the declarations into loop body instead.